interface/src/CrashHandler_Crashpad.cpp

```diff
@@ -58,29 +141,179 @@ static const QString CRASHPAD_HANDLER_NAME { "crashpad_handler" };
+// with the internal structures when operating in a 64-bit environment
+// as discovered and described here: https://stackoverflow.com/questions/39113168/c-rtti-in-a-windows-64-bit-vectoredexceptionhandler-ms-visual-studio-2015
+
+#pragma pack(push, ehdata, 4)
+
+struct PMD_internal { // internal name: _PMD (no changes, so could in theory just use the original)
+    int mdisp;
+    int pdisp;
+    int vdisp;
+};
+
+struct ThrowInfo_internal { // internal name: _ThrowInfo (changed all pointers into __int32)
+    __int32 attributes;
+    __int32 pmfnUnwind;           // 32-bit RVA
+    __int32 pForwardCompat;       // 32-bit RVA
+    __int32 pCatchableTypeArray;  // 32-bit RVA
+};
+
+struct CatchableType_internal { // internal name: _CatchableType (changed all pointers into __int32)
+    __int32 properties;
+    __int32 pType;                // 32-bit RVA
+    PMD_internal thisDisplacement;
+    __int32 sizeOrOffset;
+    __int32 copyFunction;         // 32-bit RVA
+};
+
+#pragma warning(disable : 4200)
+struct CatchableTypeArray_internal { // internal name: _CatchableTypeArray (changed all pointers into __int32)
+    int nCatchableTypes;
+    __int32 arrayOfCatchableTypes[0]; // 32-bit RVA
+};
+#pragma warning(default : 4200)
+
+#pragma pack(pop, ehdata)
+
+// everything inside this function is extremely undocumented, attempting to extract
+// the underlying C++ exception type (or at least its name) before throwing the whole
+// mess at crashpad
+// Some links describing how C++ exception handling works in an SEH context
+// (since C++ exceptions are a figment of the Microsoft compiler):
+//  - https://www.codeproject.com/Articles/175482/Compiler-Internals-How-Try-Catch-Throw-are-Interpr
+//  - https://stackoverflow.com/questions/21888076/how-to-find-the-context-record-for-user-mode-exception-on-x64
+
+static void fatalCxxException(PEXCEPTION_POINTERS pExceptionInfo) {
+    SpinLockLocker guard(crashpadAnnotationsProtect, ANNOTATION_LOCK_WEAK_ATTEMPT);
+    if (!guard.isLocked()) {
+        return;
+    }
+
+    PEXCEPTION_RECORD ExceptionRecord = pExceptionInfo->ExceptionRecord;
+    /*
+    Exception arguments for Microsoft C++ exceptions:
+    [0] signature  - magic number
+    [1] void*      - variable that is being thrown
+    [2] ThrowInfo* - description of the variable that was thrown
+    [3] HMODULE    - (64-bit only) base address that all 32bit pointers are added to
+    */
+
+    if (ExceptionRecord->NumberParameters != 4 || ExceptionRecord->ExceptionInformation[0] != MSVC_CPP_EXCEPTION_SIGNATURE) {
+        // doesn't match expected parameter counts or magic numbers
+        return;
+    }
+
+    // get the ThrowInfo struct from the exception arguments
+    ThrowInfo_internal* pThrowInfo = reinterpret_cast<ThrowInfo_internal*>(ExceptionRecord->ExceptionInformation[2]);
+    ULONG_PTR moduleBase = ExceptionRecord->ExceptionInformation[3];
+    if(moduleBase == 0 || pThrowInfo == NULL) {
```

Comment:
```suggestion
    if (moduleBase == 0 || pThrowInfo == NULL) {
```
Space after "if", here and elsewhere.